Client/Frontend/Browser/TabManager.swift

```diff
@@ -202,6 +210,33 @@ class TabManager: NSObject {
       return allTabs
     }
   }
+  
+  /// Function for adding local tabs as synced sessions
+  /// This is used when open tabs toggle is enabled in sync settings and browser constructor
+  func addRegularTabsToSyncChain() {
+    let regularTabs = tabs(withType: .regular)
+
+    syncTabsTask?.cancel()
+
+    syncTabsTask = DispatchWorkItem {
+      DispatchQueue.main.async { [weak self] in
```

Comment:
Don't do this :) 
Make the DispatchWorkItem as normal without a queue inside.
Then when you call `synctedTabsQueue.async` below, make sure you execute on main like:
`DispatchQueue.main.async(execute: task)`.

Unless there is a reason to execute on `syncedTabsQueue`, then have the task dispatch to main?
If so, ignore this comment and mark resolved.

Reply:
You are right about this. The fix is at https://github.com/brave/brave-ios/pull/5809/commits/f0fccaea2a5b34cf4cc422532a466ab1145309f0